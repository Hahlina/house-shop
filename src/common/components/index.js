export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
