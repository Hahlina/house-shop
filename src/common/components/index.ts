export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
