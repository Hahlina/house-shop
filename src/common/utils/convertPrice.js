export const convertPrice = (price) =>
    new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD'
    }).format(price || 0);
