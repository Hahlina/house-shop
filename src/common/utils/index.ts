export * from './convertPrice';
export * from './capitalizeFirstLetter';
