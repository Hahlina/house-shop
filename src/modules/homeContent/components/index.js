export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
