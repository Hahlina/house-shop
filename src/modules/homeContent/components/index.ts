export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
