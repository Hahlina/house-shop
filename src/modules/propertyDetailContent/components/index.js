export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
