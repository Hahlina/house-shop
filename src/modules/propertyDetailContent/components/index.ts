export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
