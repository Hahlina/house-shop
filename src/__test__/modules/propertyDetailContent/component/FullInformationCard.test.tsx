import { FullInformationCard } from 'modules/propertyDetailContent/components';

import { render, screen, mockPropertyDetail } from 'common/utils/test';

describe('FullInformationCard component', () => {
    it('should render description correctly', () => {
        const { description } = mockPropertyDetail;
        render(<FullInformationCard {...mockPropertyDetail} />);
        expect(screen.getByText(`Description: ${description}`)).toBeInTheDocument();
    });

    it('should display converted price in USD format', () => {
        render(<FullInformationCard {...mockPropertyDetail} />);
        expect(screen.getByText('$250,000.00')).toBeInTheDocument();
    });

    it('should take a snapshot', () => {
        const { container } = render(<FullInformationCard {...mockPropertyDetail} />);
        expect(container).toMatchSnapshot();
    });
});
