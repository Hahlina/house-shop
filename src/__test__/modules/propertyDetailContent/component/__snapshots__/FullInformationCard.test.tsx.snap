// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullInformationCard component should take a snapshot 1`] = `
<div>
  <div
    class="sc-csKJxZ imMYB"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
    >
      Cozy House near the Forest
    </h5>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FmdGoodOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m6-1.8C18 6.57 15.35 4 12 4s-6 2.57-6 6.2c0 2.34 1.95 5.44 6 9.14 4.05-3.7 6-6.8 6-9.14M12 2c4.2 0 8 3.22 8 8.2 0 3.32-2.67 7.25-8 11.8-5.33-4.55-8-8.48-8-11.8C4 5.22 7.8 2 12 2"
        />
      </svg>
      Address: 
      Green Street, 123
    </h6>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="HailOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m5-4h2c0 2.7-.93 4.41-2.3 5.5-.5.4-1.1.7-1.7.9V22h-2v-6h-2v6H9V10.1c-.3.1-.5.2-.6.3-.53.41-1.4 1.03-1.4 3.6H5c0-2.06.35-3.78 2.11-5.29C8.21 7.81 10 7 12 7s2.68-.46 3.48-1.06C15.96 5.55 17 4.76 17 2M4 16h3v6H4z"
        />
      </svg>
      Seller: 
      John Doe
    </h6>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DescriptionOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zm4 18H6V4h7v5h5z"
        />
      </svg>
      Description: 
      A charming house nestled near a beautiful forest. Perfect for nature lovers.
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body1 price css-ahj2mt-MuiTypography-root"
    >
      $250,000.00
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LocalOfferOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m21.41 11.58-9-9C12.05 2.22 11.55 2 11 2H4c-1.1 0-2 .9-2 2v7c0 .55.22 1.05.59 1.42l9 9c.36.36.86.58 1.41.58s1.05-.22 1.41-.59l7-7c.37-.36.59-.86.59-1.41s-.23-1.06-.59-1.42M13 20.01 4 11V4h7v-.01l9 9z"
        />
        <circle
          cx="6.5"
          cy="6.5"
          r="1.5"
        />
      </svg>
    </p>
  </div>
</div>
`;
